### PR TITLE
Integrate mayhem fuzzer

### PR DESCRIPTION
**Pull request by harshavamsi**
Added dockerfile, github actions, and mayhem files for all the fuzz tests to be integrated with Mayhem.

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -34,8 +34,8 @@ path = "fuzz_targets/text_parser.rs"
 test = false
 doc = false
 
-[[bin]]
```
Comment (xansec) [inline fuzz/Cargo.toml:37]: @harshavamsi , is it possible to run this target without modifications to the `Cargo.toml`? We want to merge into upstream so it's not a good idea to modify these files if we can help it.

**Review by xansec — CHANGES_REQUESTED** [review]

File: fuzz/Cargo.toml (hunk 1 above)
Comment (harshavamsi) [reply fuzz/Cargo.toml:37]: @xansec it turns out they have this additional runner in their cargo file but do not actually have this fuzz test written. Trying to execute the fuzz tests actually fails. The mayhem integration will not pass the fuzz tests.

Comment (xansec) [reply fuzz/Cargo.toml:37]: It looks like in your `Dockerfile` you run `cargo fuzz build`. You can run `cargo fuzz build <target_name1> <target_name2> ` to only build the targets in question.

Comment (harshavamsi) [reply fuzz/Cargo.toml]: yeah, you're right. I've fixed those changes and now the build passes without editing the Cargo file!

Comment (xansec) [reply fuzz/Cargo.toml]: Awesome! LGTM!

**Pull request merged**